### PR TITLE
CON-7 hotel info page

### PR DESCRIPTION
**Pull request by nikola-lisac**

### CODE REVIEW

File: con-base/src/main/java/ba/codecentric/base/service/CountryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import ba.codecentric.base.domain.Country;
+import ba.codecentric.base.repository.CountryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class CountryServiceImpl implements CountryService{
```
Comment (BranislavLazic) [inline con-base/src/main/java/ba/codecentric/base/service/CountryServiceImpl.java]: Reformat code.

File: con-master/src/main/java/ba/codecentric/master/controller/CountryController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    private CountryService countryService;
+
+    @Autowired
+    public CountryController(CountryService countryService) {
+        this.countryService = countryService;
+    }
+
+    @RequestMapping(value = "/countries", method = RequestMethod.GET)
+    public ResponseEntity<List<Country>> getCountries() {
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/CountryController.java]: Is ResponseEntity necessary here as a return type? Wouldn't you get a same response if you return only a list of countries?

File: con-master/src/main/java/ba/codecentric/master/controller/HotelController.java [hunk 3]
```diff
@@ -19,7 +23,15 @@ public HotelController(HotelService hotelService) {
     }
 
     @RequestMapping(value = "/hotel/save", method = RequestMethod.POST)
-    public void saveHotel(@RequestBody Hotel hotel) {
-        hotelService.saveHotel(hotel);
+    public ResponseEntity<Hotel> saveHotel(@RequestBody Hotel hotel) {
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/HotelController.java]: Use RESTful approach. There's no need for "/hotel/save". You'd probably want to have only "hotel" or "hotels" as a URI. Same question as in comment above. Is there a need for ResponseEntity?

File: con-master/src/main/java/ba/codecentric/master/controller/HotelController.java [hunk 4]
```diff
@@ -19,7 +23,15 @@ public HotelController(HotelService hotelService) {
...
+        return new ResponseEntity<Hotel>(hotelService.saveHotel(hotel), HttpStatus.OK);
+
+    }
+
+    @RequestMapping(value = "/hotel/get", method = RequestMethod.GET)
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/HotelController.java]: Use RESTful approach. "hotel" or "hotels" with HTTP GET method ought to be enough.

File: con-master/src/main/java/ba/codecentric/master/controller/HotelController.java [hunk 5]
```diff
@@ -19,7 +23,15 @@ public HotelController(HotelService hotelService) {
...
+    public ResponseEntity<Hotel> getHotel() {
+        List<Hotel> hotels = hotelService.getHotels();
+        return (hotels.isEmpty() ? new ResponseEntity<>(HttpStatus.OK) : new ResponseEntity<>(hotels.get(0), HttpStatus.OK));
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/HotelController.java]: Incorrect use of HTTP status code. If hotel is not found, shouldn't HTTP status code be NotFound and not OK?

File: con-master/src/main/java/ba/codecentric/master/controller/HotelController.java [hunk 6]
```diff
@@ -19,7 +23,15 @@ public HotelController(HotelService hotelService) {
...
-    public void saveHotel(@RequestBody Hotel hotel) {
-        hotelService.saveHotel(hotel);
+    public ResponseEntity<Hotel> saveHotel(@RequestBody Hotel hotel) {
+        return new ResponseEntity<Hotel>(hotelService.saveHotel(hotel), HttpStatus.OK);
+
+    }
+
+    @RequestMapping(value = "/hotel/get", method = RequestMethod.GET)
+    public ResponseEntity<Hotel> getHotel() {
+        List<Hotel> hotels = hotelService.getHotels();
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/HotelController.java]: Why do you use "select all" to fetch a single hotel? You can use SELECT 1 instead.

File: con-master/src/main/java/ba/codecentric/master/controller/HotelController.java [hunk 7]
```diff
@@ -18,8 +18,16 @@ public HotelController(HotelService hotelService) {
...
+    @RequestMapping(value = "/hotel", method = RequestMethod.POST)
+    public Hotel saveHotel(@RequestBody Hotel hotel) {
+        return hotelService.saveHotel(hotel);
+
+    }
+
+    @RequestMapping(value = "/hotel", method = RequestMethod.GET)
+    public Hotel getHotel() {
+        Hotel hotel = hotelService.getHotel();
+        return (hotel == null) ? new Hotel() : hotel;
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/HotelController.java]: Return ResponseEntity with 404 (NotFound) HTTP status if hotel does not exist.

File: con-master/src/main/java/ba/codecentric/master/controller/CountryController.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+public class CountryController {
+
+    private CountryService countryService;
+
+    @Autowired
+    public CountryController(CountryService countryService) {
+        this.countryService = countryService;
+    }
+
+    @RequestMapping(value = "/countries", method = RequestMethod.GET)
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/CountryController.java]: HTTP method is superfluous. `@RequestMapping("/countries")` is enough. GET is by default.

File: con-master/src/main/java/ba/codecentric/master/controller/HotelController.java [hunk 9]
```diff
@@ -18,8 +18,16 @@ public HotelController(HotelService hotelService) {
...
-    @RequestMapping(value = "/hotel/save", method = RequestMethod.POST)
-    public void saveHotel(@RequestBody Hotel hotel) {
-        hotelService.saveHotel(hotel);
+    @RequestMapping(value = "/hotel", method = RequestMethod.POST)
+    public Hotel saveHotel(@RequestBody Hotel hotel) {
+        return hotelService.saveHotel(hotel);
+
+    }
+
+    @RequestMapping(value = "/hotel", method = RequestMethod.GET)
```
Comment (BranislavLazic) [inline con-master/src/main/java/ba/codecentric/master/controller/HotelController.java]: Same as in comment above. No need for HTTP method.

File: dashboard-app/src/app/pages/hotel-info/controllers/HotelCtrl.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        // getting countries from database to show in select box
+        HotelService.countries().then(function (response) {
+            $scope.countries = response.data;
+        }, function (errorMessage) {
+            console.log(errorMessage);
+        });
+
+        HotelService.getHotel().then(function(response){
+            $scope.hotel = response.data;
+        },function (error) {
```
Comment (ozegster) [inline dashboard-app/src/app/pages/hotel-info/controllers/HotelCtrl.js]: 'error' is not used.

File: dashboard-app/src/app/pages/hotel-info/controllers/HotelCtrl.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        $scope.hotel = {};
+
+        // getting countries from database to show in select box
+        HotelService.countries().then(function (response) {
+            $scope.countries = response.data;
+        }, function (errorMessage) {
+            console.log(errorMessage);
+        });
+
+        HotelService.getHotel().then(function(response){
```
Comment (BranislavLazic) [inline dashboard-app/src/app/pages/hotel-info/controllers/HotelCtrl.js]: Reformat code please. Add space between statements and curly braces. Also, do you put space after `function` keyword or not? Stick to unique coding style.

File: dashboard-app/src/app/pages/hotel-info/controllers/HotelCtrl.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+            $scope.hotel = response.data;
+        },function (error) {
+            $scope.hotel={};
+        });
+
+        $scope.submit = function () {
+            HotelService.saveHotel($scope.hotel).then(function(response){
```
Comment (BranislavLazic) [inline dashboard-app/src/app/pages/hotel-info/controllers/HotelCtrl.js]: Reformat code.

**Pull request merged**